src/overlays/actors/ovl_Obj_Boyo/z_obj_boyo.c

```diff
@@ -1,40 +1,191 @@
+Actor* ObjBoyo_FindCollidedActor(ObjBoyo* this, PlayState* play, u32* index);
+
+const ActorInit Obj_Boyo_InitVars = { ACTOR_OBJ_BOYO,
+                                      ACTORCAT_PROP,
+                                      FLAGS,
+                                      OBJECT_BOYO,
+                                      sizeof(ObjBoyo),
+                                      (ActorFunc)ObjBoyo_Init,
+                                      (ActorFunc)ObjBoyo_Destroy,
+                                      (ActorFunc)ObjBoyo_Update,
+                                      (ActorFunc)ObjBoyo_Draw };
+
+typedef void (*ObjBoyoCollisionHandler)(Actor* actor_a, Actor* actor_b);
+
+typedef struct ObjBoyoUnkStruct {
+    /* 0x0 */ ObjBoyoCollisionHandler colHandler;
+    /* 0x4 */ s16 id;
+} ObjBoyoUnkStruct; // size = 0x8
+
+static AnimatedMaterial* D_06000E88;
+
+static ColliderCylinderInit gColliderCylinderInit = {
+    {
+        /* ColliderInit base */
+        COLTYPE_NONE,                                       /* u8 coltype */
+        AT_NONE,                                            /* u8 atFlags */
+        AC_TYPE_PLAYER | AC_HARD | AC_ON,                   /* u8 acFlags */
+        OC1_TYPE_2 | OC1_TYPE_1 | OC1_TYPE_PLAYER | OC1_ON, /* u8 ocFlags1 */
+        OC2_TYPE_2,                                         /* u8 ocFlags2 */
+        COLSHAPE_CYLINDER                                   /* u8 shape */
+    },
+    {
+        /* ColliderInfoInit info */
+        ELEMTYPE_UNK0, /* u8 elemType */
+        { 0x00000000 /* u32 dmgFlags */, 0x00 /* u8 effect */, 0x00 /* u8 defense */ },
+        { 0x01CBFFBE /* u32 dmgFlags */, 0x00 /* u8 effect */, 0x00 /* u8 defense */ },
+        TOUCH_NONE, /* u8 toucherFlags */
+        BUMP_ON,    /* u8 bumperFlags */
+        OCELEM_ON   /* u8 ocElemFlags */
+    },
+    { 60 /* s16 radius */, 140 /* s16 height */, 0 /* yShift */, { 0 /* s16 x */, 0 /* s16 y */, 0 /*s16 z */ } },
+};
+
+static InitChainEntry gInitChainEntry[5] = { 0xB0FC0FA0, 0xB100012C, 0xB104012C, 0x48580064,
+                                             0x00000000 }; /* unable to generate initializer */
+
+static objBoyoUnkStruct gCollisionHandling[] = {
+    { ObjBoyo_PushPlayer, 0x021D0000 }, { func_809A5DE0, 0x00090000 }, { func_809A5E14, 0x0 }, { NULL, 0x0 }
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boyo/ObjBoyo_Init.s")
+void ObjBoyo_Init(ObjBoyo* this, PlayState* play) {
+    ColliderCylinder* temp_a1;
+
+    Actor_ProcessInitChain(&this->actor, gInitChainEntry);
+    temp_a1 = &this->collider;
+    Collider_InitCylinder(play, temp_a1);
+    Collider_SetCylinder(play, temp_a1, &this->actor, &gColliderCylinderInit);
+    Collider_UpdateCylinder(&this->actor, temp_a1);
+    this->actor.colChkInfo.mass = 0xFF;
+    this->animatedMaterial = Lib_SegmentedToVirtual((void*)&D_06000E88);
+}
+
+void ObjBoyo_Destroy(ObjBoyo* thisx, PlayState* play) {
+    Collider_DestroyCylinder(play, &THIS->collider);
+}
```

Comment:
```suggestion
void ObjBoyo_Destroy(Actor* thisx, PlayState* play2) {
    PlayState* play = play2;
    ObjBoyo* this = THIS;

    Collider_DestroyCylinder(play, &this->collider);
}
```
This is interesting. To get the proper upcast, you will need to introduce a play2 as well.

Reply:
Why is that necessary in this case and not in other function ?

Reply:
For whatever reason the function does not match without the extra play temp. Try it yourself. Remove the play temp and see what asm differ says.